lib/rpcs/v1_5_0/get_node_status.rs

```diff
@@ -7,156 +7,55 @@ use casper_hashing::Digest;
-pub enum FastSync {
-    /// Fast-syncing has not started yet.
-    #[serde(rename = "not yet started")]
-    NotYetStarted,
-    /// Initial setup is being performed.
-    Starting,
-    /// Currently fetching the trusted block header.
-    FetchingTrustedBlockHeader(BlockHash),
-    /// Currently syncing the trie-store under the given state root hash.
-    #[serde(rename = "fetching_global_state_under_given_block")]
-    FetchingTries {
-        /// The height of the block containing the given state root hash.
-        block_height: u64,
-        /// The global state root hash.
-        state_root_hash: Digest,
-        /// The reason for syncing the trie-store.
-        reason: FetchingTriesReason,
-        /// The number of remaining tries to fetch (this value can rise and fall as the task
-        /// proceeds).
-        #[serde(rename = "number_of_remaining_tries_to_fetch")]
-        num_tries_to_fetch: usize,
-    },
-    /// Currently fetching the block at the given height and its deploys in preparation for
-    /// executing it.
-    #[serde(rename = "fetching_block_and_deploys_at_given_block_height_before_execution")]
-    FetchingBlockAndDeploysToExecute(u64),
-    /// Currently executing the block at the given height.
-    #[serde(rename = "executing_block_at_height")]
-    ExecutingBlock(u64),
-    /// Currently retrying the execution of the block at the given height, due to an unexpected
-    /// mismatch in the previous execution result (due to a mismatch in the deploys' approvals).
-    RetryingBlockExecution {
-        /// The height of the block being executed.
-        block_height: u64,
-        /// The retry attempt number.
-        attempt: usize,
-    },
-    /// Fast-syncing has finished, and the node will shortly transition to participating mode.
-    Finished,
-}
-
-/// The progress of a single sync-block task, many of which are performed in parallel during
-/// sync-to-genesis.
-///
-/// The task progresses from each variant to the next linearly.
-#[derive(Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
-pub enum SyncBlockFetching {
-    /// Currently fetching the block and all its deploys.
-    #[serde(rename = "block and deploys")]
-    BlockAndDeploys,
-    /// Currently syncing the trie-store under the state root hash held in the block.
-    #[serde(rename = "global_state_under_given_block")]
-    Tries {
-        /// The global state root hash.
-        state_root_hash: Digest,
-        /// The number of remaining tries to fetch (this value can rise and fall as the task
-        /// proceeds).
-        #[serde(rename = "number_of_remaining_tries_to_fetch")]
-        num_tries_to_fetch: usize,
-    },
-    /// Currently fetching the block's signatures.
-    #[serde(rename = "block signatures")]
-    BlockSignatures,
+/// The state of the reactor.
+#[derive(Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Hash, Serialize, Deserialize, Debug)]
+pub enum ReactorState {
```

Comment:
Maybe we should consider making public all of the stuff from the node that we're using in other places, like `casper-db-utils` and this repo. This code duplication will prove hard to maintain when we change things.

Reply:
Yeah, couldn't agree more.  This already is on the cards for node 2.0.